test/commands/distribute/releases/edit-test.ts

```diff
@@ -0,0 +1,218 @@
+import { expect } from "chai";
+import * as Nock from "nock";
+import * as Temp from "temp";
+import * as _ from "lodash";
+
+import EditReleaseCommand from "../../../../src/commands/distribute/releases/edit";
+import { CommandArgs, CommandResult, CommandFailedResult } from "../../../../src/util/commandline";
+
+Temp.track();
+
+describe("distribute releases edit command", async () => {
+  const fakeAppOwner = "fakeAppOwner";
+  const fakeAppName = "fakeAppName";
+  const fakeAppIdentifier = `${fakeAppOwner}/${fakeAppName}`;
+  const fakeToken = "c1o3d3e7";
+  const fakeReleaseId = "5";
+  /* tslint:disable-next-line:no-http-string */
+  const fakeHost = "http://localhost:1700";
+  const releaseIdOption = "--release-id";
+
+  before(() => {
+    Nock.disableNetConnect();
+  });
+
+  afterEach(() => {
+    Nock.cleanAll();
+  });
+
+  after(() => {
+    Nock.enableNetConnect();
+  });
+
+  context("completes successfully", () => {
+    it("if enable release", async () => {
+      // Arrange
+      const executionScope = _.flow(setupReleaseDetailsResponse, setupUpdateReleaseResponse)(Nock(fakeHost));
+
+      // Act
+      const command = new EditReleaseCommand(getCommandArgs([releaseIdOption, fakeReleaseId, "enabled"]));
+      const result = await command.execute();
+
+      testCommandSuccess(result, executionScope);
+    });
+
+    it("if disable release", async () => {
+      // Arrange
+      const executionScope = _.flow(setupReleaseDetailsResponse, setupUpdateReleaseResponse)(Nock(fakeHost));
+
+      // Act
+      const command = new EditReleaseCommand(getCommandArgs([releaseIdOption, fakeReleaseId, "disabled"]));
+      const result = await command.execute();
+
+      testCommandSuccess(result, executionScope);
+    });
+  });
+
+  context("failed", () => {
+    it("if --release-id option wasn't specified", async () => {
+      // Arrange
+      let errorMessage: string;
+      const expectedErrorMessage = "Missing required option -r / --release-id";
+
+      let command;
+      // Act
+      try {
+        command = new EditReleaseCommand(getCommandArgs([releaseIdOption]));
+      } catch (e) {
+        errorMessage = e.message;
+      }
+
+      // Assert
+      expect(errorMessage).to.eql(expectedErrorMessage);
+    });
+
+    it("if --release-id option value is not valid", async () => {
+      // Arrange
+      const expectedErrorMessage = "notanumber is not a valid release id";
+
+      // Act
+      const command = new EditReleaseCommand(getCommandArgs([releaseIdOption, "notanumber", "disabled"]));
+      const result = await command.execute();
+
+      // Assert
+      expect((result as CommandFailedResult).errorMessage).to.eql(expectedErrorMessage);
+    });
+
+    it("if state option wasn't specified", async () => {
+      // Arrange
+      let errorMessage: string;
+      const expectedErrorMessage = "Missing required positional argument State";
```

Comment:
Not your doing but a note to self - I really don't like the wording here. Using the terminology `required positional argument` seems a bit hard to understand to me.